### PR TITLE
Disks success states

### PR DESCRIPTION
**Pull request by eatonphil**

### CODE REVIEW

**Comment by mparke** [comment]
@eatonphil merge conflicts

**Comment by eatonphil** [comment]
@mparke done.

File: src/linodes/linode/settings/components/AddModal.js [hunk 1]
```diff
@@ -3,59 +3,52 @@ import React, { Component, PropTypes } from 'react';
...
-      const errors = await reduceErrors(response);
-      this.setState({ errors });
-    }
+    const { label, size, distribution, password, filesystem } = this.state;
+    const data = {
+      label,
+      filesystem,
+      size: parseInt(size),
+      distribution: distribution || null,
+      root_pass: password,
```
Comment (abemassry) [inline src/linodes/linode/settings/components/AddModal.js]: or null, spoke about in person

Comment (eatonphil) [inline src/linodes/linode/settings/components/AddModal.js]: +1 done.

**Comment by eatonphil** [comment]
thank you!

**Pull request merged**